prusti-viper/src/encoder/encoder.rs

```diff
@@ -1282,7 +1282,12 @@ impl<'v, 'tcx> Encoder<'v, 'tcx> {
             let procedure = self.env.get_procedure(wrapper_def_id);
             let pure_function_encoder =
                 PureFunctionEncoder::new(self, proc_def_id, procedure.get_mir(), false);
-            let function = if self.is_trusted(proc_def_id) {
+            let function = if let Some(predicate_body) = self.get_predicate_body(proc_def_id) {
+                let predicate_body_encoded = self.encode_assertion(predicate_body, procedure.get_mir(), None, &[], None, false, None, ErrorCtxt::GenericExpression)
+                    .run_if_err(cleanup)?;
+                pure_function_encoder.encode_predicate_function(predicate_body_encoded)
+                    .run_if_err(cleanup)?
+            } else if self.is_trusted(proc_def_id) {
```

Comment:
This makes me wonder, what happens if the user specifies both `#[predicate]` and `#[trusted]`?